pkg/kratos/service.go

```diff
@@ -159,12 +289,50 @@ func (s *Service) UpdateLoginFlow(
 	return &returnToResp, resp.Cookies(), nil
 }
 
+func (s *Service) UpdateSettingsFlow(
+	ctx context.Context, flow string, body kClient.UpdateSettingsFlowBody, cookies []*http.Cookie,
+) (*kClient.SettingsFlow, []*http.Cookie, error) {
+	ctx, span := s.tracer.Start(ctx, "kratos.Service.UpdateSettingsFlow")
+	defer span.End()
+
+	settingsFlow, resp, err := s.kratos.FrontendApi().
+		UpdateSettingsFlow(ctx).
+		Flow(flow).
+		UpdateSettingsFlowBody(body).
+		Cookie(httpHelpers.CookiesToString(cookies)).
+		Execute()
+
+	if err != nil && resp.StatusCode != http.StatusOK {
+		s.logger.Debugf("full HTTP response: %v", resp)
+		err := s.getUiError(resp.Body)
+
+		return nil, nil, err
+	}
+
+	return settingsFlow, resp.Cookies(), nil
+}
+
 func (s *Service) getUiError(responseBody io.ReadCloser) (err error) {
-	errorMessages := new(kClient.LoginFlow)
+	errorMessages := new(UiErrorMessages)
 	body, _ := io.ReadAll(responseBody)
 	json.Unmarshal([]byte(body), &errorMessages)
 
 	errorCodes := errorMessages.Ui.Messages
+
+	// if no message was found, search through nodes
+	if len(errorCodes) == 0 {
+		nodes := errorMessages.Ui.Nodes
```

Comment:
we don't perform any validation on the `errorMessages.Ui` so to play it on the safe side we should proabbly use the getter method `GetNodes` same got `node.Messages`